state/factory/factory.go

```diff
@@ -471,6 +480,9 @@ func (sf *factory) PutBlock(ctx context.Context, blk *block.Block) error {
 		)
 	}
 
+	if blk.Height() == sf.stopHeight {
+		return fmt.Errorf("factory catches up to stop height %d", sf.stopHeight)
```

Comment:
it's more accurate to say "chain sync halted right before committing block %d", sf.stopHeight

Reply:
fixed.